src/sentry/api/event_search.py

```diff
@@ -130,8 +131,8 @@ def translate(pat):
 quoted_raw_search    = spaces quoted_value spaces
 
 # standard key:val filter
-basic_filter         = negation? search_key sep search_value
-quoted_basic_filter  = negation? search_key sep quoted_value
+basic_filter         = negation? search_key sep ((open_bracket in_value (comma space* in_value)* closed_bracket) / value)
```

Comment:
nit: I think we might want to enforce that we're at the end of a line or there's a space after a square bracket since this is valid now:
`project.id:[1,2]project.id:1`
which turns into
`project.id in [1,2] and project.id=1`